### PR TITLE
CAMEL-12947:MockEndpoint.expectedHeaderReceived should fail when no e…

### PR DESCRIPTION
**Pull request by jonathanvila**
…xchange received (#2669)

### CODE REVIEW

File: components/camel-test/src/test/java/org/apache/camel/test/patterns/MockEndpointFailNoHeaderTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.Test;
+public class MockEndpointFailNoHeaderTest extends CamelTestSupport {
+
+    @EndpointInject(uri = "mock:result")
+    protected MockEndpoint resultEndpoint;
+
+    @Produce(uri = "direct:start")
+    protected ProducerTemplate template;
+
+    @Override
```
Comment (aldettinger) [inline components/camel-test/src/test/java/org/apache/camel/test/patterns/MockEndpointFailNoHeaderTest.java]: Do we really need to dump the route coverage for this test ?

**Comment by aldettinger** [comment]
I wonder if it had already been [merged](https://github.com/apache/camel/commit/4d7d8be6ff693343207fb10f515a77fb7879224b).
Moreover, the related [issue](https://issues.apache.org/jira/projects/CAMEL/issues/CAMEL-12947) seems to have been already resolved.

**Comment by oscerd** [comment]
It's a backport of the fix for 2.23.x. it is fixed on 3 and 2.x and 2.24.x

**Comment by oscerd** [comment]
There was a discussion about this

**Comment by oscerd** [comment]
Merged. Thanks

**Pull request merged**